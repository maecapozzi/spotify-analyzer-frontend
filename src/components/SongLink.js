import React from 'react'

const SongLink = (props) => (          
  <ul className='search-results__ul' key={props.result.id} onClick={props.handleClick}>
    {props.result.name} by {props.result.artist}
  </ul>
)

export default SongLink

