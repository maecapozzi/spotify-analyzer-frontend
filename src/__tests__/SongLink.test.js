import React from 'react'
import { shallow } from 'enzyme'
import { configure } from 'enzyme'
import Adapter from 'enzyme-adapter-react-15'

import SongLink from '../components/SongLink'

configure({ adapter: new Adapter() })

test('it renders correctly', () => {
  const songLinkProps = {
    result: {
      id: '6rPO02ozF3bM7NnOV4h6s2',
      name: 'sf',
      artist: 'sgs'
    },
    handleClick: () => {
      console.log('click')
    }
  }

  const component = shallow(<SongLink {...songLinkProps} />)
  expect(component).toMatchSnapshot()
})
