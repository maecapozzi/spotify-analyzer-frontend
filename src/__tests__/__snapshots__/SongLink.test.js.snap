// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SongLink
    handleClick={[Function]}
    result={
        Object {
            "artist": "sgs",
            "id": "6rPO02ozF3bM7NnOV4h6s2",
            "name": "sf",
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": "6rPO02ozF3bM7NnOV4h6s2",
    "nodeType": "host",
    "props": Object {
      "children": Array [
        "sf",
        " by ",
        "sgs",
      ],
      "className": "search-results__ul",
      "onClick": [Function],
    },
    "ref": null,
    "rendered": Array [
      "sf",
      " by ",
      "sgs",
    ],
    "type": "ul",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": "6rPO02ozF3bM7NnOV4h6s2",
      "nodeType": "host",
      "props": Object {
        "children": Array [
          "sf",
          " by ",
          "sgs",
        ],
        "className": "search-results__ul",
        "onClick": [Function],
      },
      "ref": null,
      "rendered": Array [
        "sf",
        " by ",
        "sgs",
      ],
      "type": "ul",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
